src/lammpsio/topology.py

```diff
@@ -0,0 +1,111 @@
+import numpy
+
+
+class Topology:
+    def __init__(self, N, num_members):
+        self._N = int(N)
+        self._num_members = num_members
+
+        self._id = None
+        self._typeid = None
+        self._members = None
+
+    @property
+    def N(self):
+        return self._N
+
+    @property
+    def id(self):
+        """:class:`numpy.ndarray`: Bond IDs."""
+        if not self.has_id():
+            self._id = numpy.arange(1, self.N + 1)
+        return self._id
+
+    @id.setter
+    def id(self, value):
+        if value is not None:
+            v = numpy.array(value, ndmin=1, copy=False, dtype=int)
+            if v.shape != (self.N,):
+                raise TypeError("Ids must be a size N array")
+            if not self.has_id():
+                self._id = numpy.arange(1, self.N + 1)
+            numpy.copyto(self._id, v)
+        else:
+            self._id = None
+
+    def has_id(self):
+        """Check if configuration has connection IDs.
+
+        Returns
+        -------
+        bool
+            True if connection IDs have been initialized.
+
+        """
+        return self._id is not None
+
+    @property
+    def typeid(self):
+        """:class:`numpy.ndarray`: Bond typeids."""
+        if not self.has_typeid():
+            self._typeid = numpy.zeros(self.N, dtype=int)
```

Comment:
Check the snapshot, but I think this should probably be 1s as a default since the LAMMPS types start from 1 (not 0).
```suggestion
            self._typeid = numpy.ones(self.N, dtype=int)
```